src/shieldpy/watertank.py

```diff
@@ -0,0 +1,21 @@
+import nfa
+from typing import NewType
+
+# Water Tank Example from Figure 4
+# TODO do we want figure 4 or 5?
+# TODO move this to something like examples/
+
+Switch = Enum('Switch', ['OPEN', 'CLOSE'])
+
+Level = NewType('Level', int) # Would be cool if I can bound this between 0 and 100
+minLevel = Level(0)
+maxLevel = Level(100)
```

Comment:
yeah there are some design patterns for this (maybe callable validators, clsmethods, smart constructors), unclear if bothering to implement is correct. research code isn't prod code, after all.